Set reading to float

truongdd03 opened the pull request:
# Description

Testing found that the space and access time are acceptable when changing the reading type from real (32-bits) to float (64-bit). This PR makes this change for the reading table and functions using these values. DB migration is also created. I have done a full migration and it works as expected on my machine.

Fixes #650 

## Type of change

- [ ] Note merging this changes the node modules
- [x] Note merging this changes the database configuration.
- [ ] This change requires a documentation update

## Checklist

- [x] I have followed the [OED pull request](https://openenergydashboard.github.io/developer/pr.html) ideas
- [x] I have removed text in ( ) from the issue request

## Limitations

truongdd03 committed:
Change reading to float

truongdd03 committed:
Put back comments

huss commented:
I just want to leave a record that I searched all SQL uses of real and found these:

![Screen Shot 2022-05-10 at 8 47 59 AM](https://user-images.githubusercontent.com/987062/167644231-eb10bdef-aad9-43fb-8344-15f3266ff808.png)

Given their usage they are probably fine at 32-bit so they are not being changed now.

huss committed:
comments and drop functions before create

- If you already had the compare functions then they must be dropped
before created since signature changed. I added the drop just to be
careful.
- Added comments to migration files about dropping & refreshing views
if doing manually.

huss reviewed (approved):
This works fine and as intended. I did not do a full migration but tested each migration individually. I push a few small changes to clarify about order and needed operations before alters/creates. Can @truongdd03 look at them to make sure they are okay?

I note that I realized you have to refresh the reading views after the migration since there are new views. That will be dealt with separately.

I think this is ready to merge after my changes are checked.

truongdd03 commented:
Thank you for your review and addition, @huss. New changes look good to me so I think the PR can be merged now.
